### PR TITLE
Introduce build-component-images as step to post-monorepo joy

### PR DESCRIPTION
**Pull request by jchesterpivotal**
Pretty narrowly focused on just the minimum here. I expect I'll be able to follow with PRs to tear out old jobs, tasks and resources.

### CODE REVIEW

File: pipelines/riff.yml [hunk 1]
```diff
@@ -460,6 +476,89 @@ jobs: ##########################################################################
...
+      params:
+        pre: build
+    - put: github-commit-status-riff
+      params:
+        commit: git-riff
+        state: pending
+        description: Building, testing and creating docker images for riff server components
+
+  - aggregate: # build and test
+    - task: test-topic-controller
```
[the remark this answers is not in the document]
Comment (ericbottard) [reply pipelines/riff.yml]: Ah, scratch that. I see docker push is after

File: pipelines/riff.yml [hunk 2]
```diff
@@ -460,6 +476,89 @@ jobs: ##########################################################################
...
+      file: git-pfs-ci/tasks/riff-components/test/task.yml
```
Comment (ericbottard) [inline pipelines/riff.yml]: I thought we agreed on not trying to factorize behavior across components.
I believe this PR will work, but then we should at least clean unused files

Comment (jchesterpivotal) [reply pipelines/riff.yml]: Only sidecar needs the kafka container, all the others work with the `test` task. I did notice that while I updated that task (`test-function-sidecar`) in the `riff` pipeline, it still have a `params` block hanging off from when I tried to do it using the factorized `test` task.

I had in mind to push the cleanup into a following commit. Not that I don't want to do it, but I'd like to get to the working case first.

**Comment by ericbottard** [comment]
LGTM then.
I am about to merge this, is this ok ?

Also, I noticed you used the projectriff/ci repo to issue your PR. We usually keep those "reference" repos clean and issue PRs from forks

**Comment by jchesterpivotal** [comment]
Yeah, I misstepped on the workflow. Happy for the merge to go ahead.

**Pull request merged**